### ISSUE TITLE
Allow ignore UnusedPrivateMember for annotated methods

### ISSUE THREAD

**Issue by osipxd** [detekt/detekt#2660]
## Expected Behavior of the rule

Is it possible to ignore UnusedPrivateMember rule for members annotated with specified annotations?

Example of config:
```yml
UnusedPrivateMember:
    active: true
    allowedNames: "(_|ignored|expected|serialVersionUID)"
    ignoreAnnotatedWith: "androidx.lifecycle.OnLifecycleEvent" # <--
```

Example of code:
```kotlin
@OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
private fun onResume() { // <-- OK
    // ...
}
```

## Context

Some event-based frameworks can use reflection to access annotated methods. Some of the frameworks also allow these methods to be private. In this case, methods actually used but Detekt don't know about it.

[AndroidX Lifecycle](https://developer.android.com/topic/libraries/architecture/lifecycle#lc)

**Comment by schalkms** [comment]
duplicate of #2231
Please follow the discussion in the linked issue.

**Comment by schalkms** [comment]
duplicate of #2231
Thanks for the feedback btw! Please follow the discussion in the linked issue and give a vote for this feature.
